std/bigint.d

```diff
@@ -566,6 +566,71 @@ public:
     }
 
     /**
+        Implements casting to integer types.
+
+        Throws: $(XREF conv,ConvOverflowException) if the number exceeds
+        the target type's range.
+     */
+    T opCast(T:ulong)() /*pure*/ const
+    {
+        if (isUnsigned!T && sign)
+            { /* throw */ }
+        else
+        if (data.ulongLength == 1)
+        {
+            ulong l = data.peekUlong(0);
+            if (l <= T.max)
+                return cast(T)(sign ? -long(l) : l);
+            static if (isSigned!T)
+                if (sign && l == ulong(T.max)+1)
+                    return T.min;
+        }
+
+        import std.conv : ConvOverflowException;
+        import std.string : format;
+        throw new ConvOverflowException(
+            "BigInt(%d) cannot be represented as a %s"
+            .format(this, T.stringof));
+    }
+
+    ///
+    unittest
```

Comment:
This unit test will become the example for `to!ulong`, but it doesn't have a call to to!ulong, just calls for int and ubyte.